### PR TITLE
Refactor provider mailer specs

### PR DESCRIPTION
**Pull request by C-gyorfi**
## Context

The mailer specs are long and we want to refactor them, the candidate mailer is 700 lines long and it has some confusing setup

## Changes proposed in this pull request

This PR refactors the following mailer specs:
- Adds a shared example to provider mailer spec to dry out the tests
- Adds shared example to candidate mailer and cut down some of the duplications

## Guidance to review

I think the refactoring works in `ProviderMailer` specs, I managed to cut down half of the code and I believe its quite readable. However, `CandidateMailer` specs are much more challenging to handle, most mails has different parameter signature or require more setup which makes it hard to fit them in shared examples. I managed to cut down some code, but I am not sure if it helps. What do you think?

## Link to Trello card

https://trello.com/c/Igu4bzNf/132-refactor-tests-or-mailers-to-make-mailer-specs-nicer-medium

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: spec/mailers/provider_mailer_spec.rb [hunk 1]
```diff
@@ -3,214 +3,112 @@
...
-      expect(@mail.body.encoded).to include("Dear #{@provider_user.full_name}")
+      expect(email.subject).to include(subject)
     end
 
-    it 'includes a link to the provider sign in page' do
-      expect(@mail.body.encoded).to include(provider_interface_sign_in_path)
+    content.each do |key, expectation|
+      it "sends an email containing the #{key} in the body" do
+        expect(email.body).to include(expectation)
+      end
```
Comment (chubberlisk) [inline spec/mailers/provider_mailer_spec.rb]: Outputs like the below, reads well!

```
ProviderMailer
  .declined
    behaves like a provider mail with subject and content
      sends an email with the correct subject
      sends an email containing the candidate name in the body
      sends an email containing the course name and code in the body
      sends an email containing the provider name in the body
```

File: spec/mailers/provider_mailer_spec.rb [hunk 2]
```diff
@@ -3,214 +3,112 @@
...
+    @application_choice.application_form.update(submitted_at: Time.zone.now - 5.days)
+  end
 
-    it 'includes the reject by default days' do
-      expect(@mail.body.encoded).to include('after 123 working days')
-    end
+  describe 'Send account created email' do
+    it_behaves_like('a provider mail with subject and content', :account_created,
+                    I18n.t!('provider_account_created.email.subject'),
+                    'provider name' => "Dear Johny English\r\n",
```
Comment (chubberlisk) [inline spec/mailers/provider_mailer_spec.rb]: Do we need the `\r\n`?

File: spec/mailers/provider_mailer_spec.rb [hunk 3]
```diff
@@ -3,214 +3,112 @@
...
+  end
 
-    it 'includes a link to the application' do
-      expect(@mail.body.encoded).to include(provider_interface_application_choice_url(application_choice))
-    end
+  describe 'Send application submitted email' do
+    it_behaves_like('a provider mail with subject and content', :application_submitted,
+                    I18n.t!('provider_application_submitted.email.subject',
+                            course_name_and_code: 'Computer Science (6IND)'),
+                    'provider name' => "Dear Johny English\r\n",
```
Comment (chubberlisk) [inline spec/mailers/provider_mailer_spec.rb]: Same here, do we need the `\r\n`?

File: spec/mailers/provider_mailer_spec.rb [hunk 4]
```diff
@@ -3,214 +3,112 @@
...
-      rbd_date = application_choice.reject_by_default_at
-      expect(@mail.body.encoded).to include("by #{rbd_date.to_s(:govuk_date).strip}")
-    end
+    it_behaves_like('a provider mail with subject and content', :chase_provider_decision,
+                    I18n.t!('provider_application_waiting_for_decision.email.subject',
+                            candidate_name: 'Harry Potter'),
+                    'provider name' => 'Dear Johny English',
+                    'candidate name' => 'Harry Potter',
+                    'course name and code' => 'Computer Science (6IND)',
+                    'time to responde' => "Only #{TimeLimitConfig.limits_for(:chase_provider_before_rbd).first.limit} working days left to respond",
```
Comment (chubberlisk) [inline spec/mailers/provider_mailer_spec.rb]:
```suggestion
                    'time to respond' => "Only #{TimeLimitConfig.limits_for(:chase_provider_before_rbd).first.limit} working days left to respond",
```

File: spec/mailers/provider_mailer_spec.rb (hunk 3 above)
Comment (C-gyorfi) [reply spec/mailers/provider_mailer_spec.rb]: I'm pretty sure I had a failing test because of this, but now I tried and no, we don't need 👌

**Comment by C-gyorfi** [comment]
As discussed with @tvararu, I'll merge the ProviderMailer spec and the CandidateMailer spec needs more love, potentially refactoring the mailer itself. I'll do them in multiple smoller slices 📄

**Pull request merged**